src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -215,6 +227,14 @@ async Task ConsumeDelayedMessages(ReceiveMessageRequest request, CancellationTok
                                 sendMessageRequest.DelaySeconds = Convert.ToInt32(remainingDelay);
                             }
                         }
+                        
+                        // because message attributes are part of the content size restriction we want to prevent message size from changing thus we add it 
+                        // for native delayed deliver as well
+                        sendMessageRequest.MessageAttributes[Headers.MessageId] = new MessageAttributeValue
+                        {
+                            StringValue = messageId,
```

Comment:
This will be null if the orignal message didn't have the messageId header set, but it might be safe to assume delayed messages have all been put there by NServiceBus code and will have that header?

Reply:
Addressed